source/slang/slang-compiler.cpp

```diff
@@ -2197,24 +2197,23 @@ SlangResult dissassembleDXILUsingDXC(
     }
 
     CompileResult& TargetProgram::_createWholeProgramResult(
-        const List<Int>&        entryPointIndices,
         BackEndCompileRequest*  backEndRequest,
         EndToEndCompileRequest* endToEndRequest)
     {
-        for (auto entryPointIndex = entryPointIndices.begin(); entryPointIndex != entryPointIndices.end(); entryPointIndex++) {
-            if (*entryPointIndex >= m_entryPointResults.getCount())
-                m_entryPointResults.setCount(*entryPointIndex + 1);
+        // It is possible that entry points goot added to the `Program`
+        // *after* we created this `TargetProgram`, so there might be
+        // a request for an entry point that we didn't allocate space for.
+        //
+        // TODO: Change the construction logic so that a `Program` is
+        // constructed all at once rather than incrementally, to avoid
+        // this problem.
+        //
```

Comment:
Note: this comment seems to have been copy-pasted over from `_createEntryPointResult`, and doesn't seem accurate to anything that is going on here. It might be good to change it to a more accurate comment about how the current codegen logic requires a list of entry point indices to generate, so that a "whole-program" result requires us to manually specify all entry points for inclusion.

Reply:
In the context of talking about specialization and linking, it is worth noting that the list of entry points here is *not* exactly the same as the list of entry points declared via `[shader(...)]` attributes in the input.

What is being collected here is a list of all entry points that were linked into `m_program`, which could include explicitly-specialized entry points if this `TargetProgram` was created via the `IComponentType` part of the API.

For the older C-style API, the `m_program` will always consist of all referenced modules, plus all the entry points that were declared/found, each specialized to any specialization arguments that were specified via API/command-line.